doc/qos/mpls_exp_to_tc_map.md

```diff
@@ -0,0 +1,131 @@
+# MPLS EXP to TC map
+
+## 1. Table of Content
+
+- [MPLS EXP to TC map](#mpls-exp-to-tc-map)
+  - [1. Table of Content](#1-table-of-content)
+  - [2. Revision](#2-revision)
+  - [3. Scope](#3-scope)
+  - [4. Definitions/Abbreviations](#4-definitionsabbreviations)
+  - [5. Overview](#5-overview)
+  - [6. Requirements](#6-requirements)
+  - [7. Architecture Design](#7-architecture-design)
+  - [8. High-Level Design](#8-high-level-design)
+    - [8.1. DB](#81-db)
+    - [8.2. sonic-swss-common](#82-sonic-swss-common)
+    - [8.3. sonic-swss](#83-sonic-swss)
+    - [8.4. sonic-utilities](#84-sonic-utilities)
+    - [8.5. Other implications](#85-other-implications)
+  - [9. SAI API](#9-sai-api)
+  - [10. Configuration and management](#10-configuration-and-management)
+    - [10.1. CLI/YANG model Enhancements](#101-cliyang-model-enhancements)
+    - [10.2. Config DB Enhancements](#102-config-db-enhancements)
+  - [11. Warmboot and Fastboot Design Impact](#11-warmboot-and-fastboot-design-impact)
+  - [12. Restrictions/Limitations](#12-restrictionslimitations)
+  - [13. Testing Requirements/Design](#13-testing-requirementsdesign)
+    - [13.1. Unit Test cases](#131-unit-test-cases)
+    - [13.2. System Test cases](#132-system-test-cases)
+  - [14. Open/Action items - if any](#14-openaction-items---if-any)
+
+## 2. Revision
+
+| Rev |     Date    |       Author            | Change Description                         |
+|:---:|:-----------:|:-----------------------:|--------------------------------------------|
+| 0.1 | 16/08/2021  |     Alexandru Banu      | Initial version                            |
+
+## 3. Scope
+
+This HLD extends SONiC to support MPLS EXP to TC mappings.
+
+## 4. Definitions/Abbreviations
+
+TC = Traffic Class
+QoS = Quality of Service
+
+## 5. Overview
+
+This new enhancement adds support to SONiC for MPLS EXP to TC map which allows QoS to work on MPLS packets.
+
+## 6. Requirements
+
+User can configure MPLS EXP to TC map at start-of-day via configuration file. CLI support will exist to offer the same amount of support as for DSCP to TC map.
+
+## 7. Architecture Design
+
+The overall SONiC architecture will not be changed and no new sub-modules will be introduced.
+
+## 8. High-Level Design
+
+### 8.1. DB
+
+The CONFIG DB will be updated to include a new "MPLS_EXP_TO_TC_MAP_TABLE" similar to the existing "DSCP_TO_TC_MAP_TABLE". This will have the following format:
```

Comment:
We're changing the DSCP_TO_TC_MAP_TABLE table representation in the config-DB now, please refer to the below PR and ensure to follow the same model for MPLS TC to TC MAP table as well.
https://github.com/Azure/sonic-buildimage/pull/7375/files

Reply:
Is there any expected date when that PR is going to get in? We'd want the MPSL TC to TC MAP PR to get in for the November release. If the referenced PR isn't going to be merged soon or at least the existing comments to be addressed in order to come to a conclusion about how the map/list structure should look like, we could proceed with the existing design and come back and update it when that PR is also merged.

Reply:
Following up on the today's meeting for clarity - is the existing `DSCP_TO_TC_MAP` table structure an issue? If not, the new table is going to look exactly the same for MPLS. If it is an issue, could you please explain how you think the table structure should be updated?

Reply:
Our code now follows the new model